File: LeetcodeProblems/blind-75/binary/missing-number/solution.js
// Missing Number - LeetCode #268

/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let xor = 0;
    for (let i = 0; i <= nums.length; i++) {
        xor ^= i;
    }
    for (let num of nums) {
        xor ^= num;
    }
    return xor;
};

File: LeetcodeProblems/blind-75/binary/missing-number/solution.py
# Missing Number - LeetCode #268

from typing import List

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        xor = 0
        for i in range(len(nums) + 1):
            xor ^= i
        for num in nums:
            xor ^= num
        return xor
